src/core/router/atoms/currentLocation.ts

```
@@ -1,11 +1,14 @@
 import { createAtom } from '~utils/atoms';
+import { dispatchMetricsEvent } from '~core/metrics/dispatch';
 
 export const currentLocationAtom = createAtom(
   {
     set: (location: Location) => location,
   },
   ({ onAction }, state = globalThis.location) => {
     onAction('set', (location) => (state = location));
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)